homeassistant/components/unifiled/light.py

```diff
@@ -1,49 +1,19 @@
 """Support for Unifi Led lights."""
 import logging
 
-from unifiled import unifiled
-import voluptuous as vol
-
-from homeassistant.components.light import (
-    ATTR_BRIGHTNESS,
-    PLATFORM_SCHEMA,
-    SUPPORT_BRIGHTNESS,
-    Light,
-)
-from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_PORT, CONF_USERNAME
-import homeassistant.helpers.config_validation as cv
+from homeassistant.components.light import ATTR_BRIGHTNESS, SUPPORT_BRIGHTNESS, Light
 
-_LOGGER = logging.getLogger(__name__)
+from .const import DOMAIN, KEY_API
 
-# Validation of the user's configuration
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_HOST): cv.string,
-        vol.Required(CONF_USERNAME): cv.string,
-        vol.Required(CONF_PASSWORD): cv.string,
-        vol.Optional(CONF_PORT, default=20443): vol.All(cv.port, cv.string),
-    }
-)
+_LOGGER = logging.getLogger(__name__)
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
+async def async_setup_entry(hass, config_entry, async_add_devices):
     """Set up the Unifi LED platform."""
 
-    # Assign configuration variables.
-    # The configuration check takes care they are present.
-    host = config[CONF_HOST]
-    port = config[CONF_PORT]
-    username = config[CONF_USERNAME]
-    password = config[CONF_PASSWORD]
-
-    api = unifiled(host, port, username=username, password=password)
+    api = hass.data[KEY_API][config_entry.entry_id]
 
-    # Verify that passed in configuration works
-    if not api.getloginstate():
-        _LOGGER.error("Could not connect to unifiled controller")
-        return
-
-    add_entities(UnifiLedLight(light, api) for light in api.getlights())
+    async_add_devices(UnifiLedLight(light, api) for light in api.getlights())
```

Comment:
Getting the lights does a request. We need to use `hass.async_add_executor_job` for that call.